diff --git a/src/better_sqlite3.cpp b/src/better_sqlite3.cpp
index 13f054a641f63dda3ebbf71ea5613cf12b054fad..c50992a8f0d223a40c10a4bc438d8a61d065f504 100644
--- a/src/better_sqlite3.cpp
+++ b/src/better_sqlite3.cpp
@@ -57,7 +57,7 @@ NODE_MODULE_INIT(/* exports, context */) {
 
 	// Initialize addon instance.
 	Addon* addon = new Addon(isolate);
-	v8::Local<v8::External> data = v8::External::New(isolate, addon);
+	v8::Local<v8::External> data = v8::External::New(isolate, addon, v8::kExternalPointerTypeTagDefault);
 	node::AddEnvironmentCleanupHook(isolate, Addon::Cleanup, addon);
 
 	// Create and export native-backed classes and functions.
diff --git a/src/util/helpers.cpp b/src/util/helpers.cpp
index 1f4b246883add58c18d8b1ae1c68d876fcee30eb..943d4d6a76b1e0cb157962773abdfbd4f77960aa 100644
--- a/src/util/helpers.cpp
+++ b/src/util/helpers.cpp
@@ -89,7 +89,7 @@ void SetPrototypeGetter(
 	recv->InstanceTemplate()->SetNativeDataProperty(
 		InternalizedFromLatin1(isolate, name),
 		func,
-		0,
+		static_cast<v8::AccessorNameSetterCallback>(nullptr),
 		data
 	);
 }
diff --git a/src/util/macros.cpp b/src/util/macros.cpp
index e75d53763e575cb1ec5f560d1303100ebb380249..a6d85b756d290ba9b06a0ae003d9a39219e7bfb7 100644
--- a/src/util/macros.cpp
+++ b/src/util/macros.cpp
@@ -27,7 +27,7 @@
 #define EasyIsolate v8::Isolate* isolate = v8::Isolate::GetCurrent()
 #define OnlyIsolate info.GetIsolate()
 #define OnlyContext isolate->GetCurrentContext()
-#define OnlyAddon static_cast<Addon*>(info.Data().As<v8::External>()->Value())
+#define OnlyAddon static_cast<Addon*>(info.Data().As<v8::External>()->Value(v8::kExternalPointerTypeTagDefault))
 #define UseIsolate v8::Isolate* isolate = OnlyIsolate
 #define UseContext v8::Local<v8::Context> ctx = OnlyContext
 #define UseAddon Addon* addon = OnlyAddon
